regression: NXP: drivers: clock_control_mcux_syscon: invalid clock source options

https://github.com/zephyrproject-rtos/zephyr/pull/77247 introduced a regression issue on some NXP platforms.  That PR added code for the MCXN series, but does not prevent it from building on other platforms.  For example, building for the board `frdm_rw612` fails if the `rtc` node is enabled.

To reproduce the errors building for the `frdm_rw612`, add this overlay:
```
&rtc {
	status = "okay";
};
```

Building the blinky sample with that overlay generates these build errors:
```
zephyr/drivers/clock_control/clock_control_mcux_syscon.c: In function 'mcux_lpc_syscon_clock_control_on':
zephyr/drivers/clock_control/clock_control_mcux_syscon.c:132:9: warning: implicit declaration of function 'CLOCK_SetupClk16KClocking' [-Wimplicit-function-declaration]
  132 |         CLOCK_SetupClk16KClocking(kCLOCK_Clk16KToVbat | kCLOCK_Clk16KToMain);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~
zephyr/drivers/clock_control/clock_control_mcux_syscon.c:132:35: error: 'kCLOCK_Clk16KToVbat' undeclared (first use in this function)
  132 |         CLOCK_SetupClk16KClocking(kCLOCK_Clk16KToVbat | kCLOCK_Clk16KToMain);
      |                                   ^~~~~~~~~~~~~~~~~~~
zephyr/drivers/clock_control/clock_control_mcux_syscon.c:132:35: note: each undeclared identifier is reported only once for each function it appears in
zephyr/drivers/clock_control/clock_control_mcux_syscon.c:132:57: error: 'kCLOCK_Clk16KToMain' undeclared (first use in this function)
  132 |         CLOCK_SetupClk16KClocking(kCLOCK_Clk16KToVbat | kCLOCK_Clk16KToMain);
      |                                                         ^~~~~~~~~~~~~~~~~~~
zephyr/drivers/clock_control/clock_control_mcux_syscon.c:137:27: error: 'kCLOCK_Rtc0' undeclared (first use in this function); did you mean 'kCLOCK_Rtc'?
  137 |         CLOCK_EnableClock(kCLOCK_Rtc0);
      |                           ^~~~~~~~~~~
      |                           kCLOCK_Rtc
```

--- a/drivers/clock_control/clock_control_mcux_syscon.c
+++ b/drivers/clock_control/clock_control_mcux_syscon.c
@@ -122,11 +122,9 @@ static int mcux_lpc_syscon_clock_control_on(const struct device *dev,
 #endif
 
 #if DT_NODE_HAS_STATUS(DT_NODELABEL(rtc), okay)
-#if CONFIG_SOC_SERIES_IMXRT5XX
+#if defined(CONFIG_SOC_SERIES_IMXRT5XX) || defined(CONFIG_SOC_SERIES_IMXRT6XX)
 	CLOCK_EnableOsc32K(true);
-#elif CONFIG_SOC_SERIES_IMXRT6XX
-	/* No configuration */
-#else /* !CONFIG_SOC_SERIES_IMXRT5XX | !CONFIG_SOC_SERIES_IMXRT6XX */
+#elif CONFIG_SOC_SERIES_MCXN
 /* 0x0 Clock Select Value Set IRTC to use FRO 16K Clk */
 #if DT_PROP(DT_NODELABEL(rtc), clock_select) == 0x0
 	CLOCK_SetupClk16KClocking(kCLOCK_Clk16KToVbat | kCLOCK_Clk16KToMain);
@@ -135,7 +133,7 @@ static int mcux_lpc_syscon_clock_control_on(const struct device *dev,
 	CLOCK_SetupOsc32KClocking(kCLOCK_Osc32kToVbat | kCLOCK_Osc32kToMain);
 #endif /* DT_PROP(DT_NODELABEL(rtc), clock_select) */
 	CLOCK_EnableClock(kCLOCK_Rtc0);
-#endif /* CONFIG_SOC_SERIES_IMXRT5XX */
+#endif /* CONFIG_SOC_SERIES_MCXN */
 #endif /* DT_NODE_HAS_STATUS(DT_NODELABEL(rtc), okay) */
 
 	return 0;

